From 321186b979973d26029470b9ea1db1d4cca74a02 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 9 Feb 2023 22:28:43 +1100
Subject: [PATCH 8/8] try HOST_PREFIX

---
 tools/setup_helpers/extension.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tools/setup_helpers/extension.py b/tools/setup_helpers/extension.py
index dd17f82a..a879de9f 100644
--- a/tools/setup_helpers/extension.py
+++ b/tools/setup_helpers/extension.py
@@ -68,11 +68,11 @@ class CMakeBuild(build_ext):
         cmake_args = [
             "-GNinja",
             f"-DCMAKE_BUILD_TYPE={cfg}",
-            f"-DCMAKE_PREFIX_PATH={os.environ['PREFIX']}",
-            f"-DCMAKE_INSTALL_PREFIX={os.environ['PREFIX']}/lib/python{dotted_pyver}/site-packages/torchtext",
+            f"-DCMAKE_PREFIX_PATH={os.environ['HOST_PREFIX']}",
+            f"-DCMAKE_INSTALL_PREFIX={os.environ['HOST_PREFIX']}/lib/python{dotted_pyver}/site-packages/torchtext",
             "-DCMAKE_VERBOSE_MAKEFILE=ON",
             f"-DPython_INCLUDE_DIR={distutils.sysconfig.get_python_inc()}",
-            f"-DTORCH_INSTALL_PREFIX={os.environ['PREFIX']}/lib/python{dotted_pyver}/site-packages/torch",
+            f"-DTORCH_INSTALL_PREFIX={os.environ['HOST_PREFIX']}/lib/python{dotted_pyver}/site-packages/torch",
             "-DBUILD_TORCHTEXT_PYTHON_EXTENSION:BOOL=ON",
             "-DRE2_BUILD_TESTING:BOOL=OFF",
             "-DBUILD_TESTING:BOOL=OFF",
-- 
2.38.1.windows.1

